### PR TITLE
Trees: use Term.ArgClause, not List[Term]

### PR DESCRIPTION
**Pull request by kitbellew**
Helps with #2869.

### CODE REVIEW

**Review by tgodzik — APPROVED** [review]
Some smaller comments, otherwise LGTM

File: scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala [hunk 1]
```diff
@@ -610,9 +617,10 @@ object Ctor {
 
 // NOTE: The name here is always Name.Anonymous.
 // See comments to Ctor.Primary and Ctor.Secondary for justification.
-@ast class Init(tpe: Type, name: Name, argss: List[List[Term]]) extends Ref {
+@ast class Init(tpe: Type, name: Name, argClauses: Seq[Term.ArgClause]) extends Ref {
   checkFields(tpe.isConstructable)
   checkParent(ParentChecks.Init)
+  @replacedField("4.6.0") final def argss: List[List[Term]] = argClauses.map(_.values).toList
```
[the remark this answers is not in the document]
Comment (kitbellew) [reply scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala:623]: the replacedField annotation macro adds deprecation warning. i can double-check, though.

[the remark this answers is not in the document]
Comment (tgodzik) [reply scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala:623]: I haven't seen them being deprecated when I checked locally released version 🤔

[the remark this answers is not in the document]
Comment (kitbellew) [reply scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala:615]: you are right, of course. fixed in #2960.

**Pull request merged**